File: algorithms/python/hammingDistance/hammingDistance.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/hamming-distance/
# Author: Miao Zhang
# Date:   2021-02-14

class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        return bin(x ^ y).count('1')

File: algorithms/cpp/hammingDistance/hammingDistance.cpp
// Source: https://leetcode.com/problems/hamming-distance/
// Author: Miao Zhang
// Date:   2021-02-14

class Solution {
public:
    int hammingDistance(int x, int y) {
        int t = x ^ y;
        int res = 0;
        while (t) {
            res += t & 1;
            t >>= 1;
        }
        return res;
    }
};
